analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+
+cellranger_cell_qc <- cellranger_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+# combine all the data frames into one
+cell_qc <- dplyr::bind_rows(
+  alevin = alevin_cell_qc,
+  alevin_fry = alevin_fry_cell_qc,
+  alevin_fry_sketch = alevin_fry_sketch_cell_qc,
+  kallisto = kallisto_cell_qc,
+  cellranger = cellranger_cell_qc,
+  .id = "tool"
+) %>%
+  dplyr::left_join(quant_info,
+                   by = c("tool" = "tool", 
+                          "quant_id" = "quant_dir"))
+```
+
+
+
+## Comparison of QC Metrics across all tools
+
+### Per Cell QC
+
+Let's first start by taking a look at some general information across all of our tools and how consistent each of these metrics is for all four of the samples. Here are the metrics we want to look at: 
+
+  * Total number of cells detected
+  * Mitochondrial content/ cell
+  * Number of UMIs/cell
+  * Number of Genes/cell 
+
+
+```{r}
+## make new data frame with mean of cells detected and standard dev
+cells_detected_stats <- cell_qc %>%
+  group_by(sample, tool) %>%
+  summarise(cells_detected = mean(cells_detected)) %>%
+  group_by(tool) %>%
+  summarise(mean = mean(cells_detected),
+            standard_error = sd(cells_detected)/sqrt(length(cells_detected)))
+
+## first look at cells detected
+ggplot(cells_detected_stats, aes(x = tool,y = mean)) +
+  geom_bar(stat = "identity", fill = "darkblue") +
+  geom_errorbar(aes(x = tool, ymin=mean-standard_error, 
+                    ymax = mean + standard_error), 
+                width = 0.4, colour = "black", alpha = 0.9, size = 1.3)+
+  theme_classic() +
+  ylab("Total Cells Detected") + 
+  xlab("")
+```
+Based on this, we can see that Cellranger is detecting what looks like the most number of cells with Alevin the least number of cells. However, Cellranger also looks like it has the largest variation, and all of the tools appear to have quite a bit of variation. It does make sense to have variation because it's likely that samples just have lower cells than other samples. Let's look at a plot that may be a little more informative and break down the number of cells detected/sample across all the tools. 
+
+
+```{r}
+cells_detected_stats <- cell_qc %>%
+  group_by(sample, tool) %>%
+  summarise(cells_detected = mean(cells_detected))
+
+## now break it down by sample 
+ggplot(cells_detected_stats, aes(x = tool, y = cells_detected)) + 
+  geom_bar(stat = "identity") + 
+  facet_wrap(~sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Number of Cells Detected") + 
+  xlab("")
+```
+Okay, now we can see that even within one sample there is quite the variation across the tools. This seems a bit concerning... Cellranger looks like it detects the most amount of cells for 2 of the 4 samples and Alevin for the other 2. But then in the 2 samples (126 and 127) that Cellranger has the most, Alevin has the least? Is there something potentially wrong with those runs with alevin? 
```

Comment:
One other thing to note is that the kallisto "cells detected" is actually being determined by the `DropletUtils::barcodeRanks` tool in the previous script. Overall, it definitely seems like this cell count metric is a bit fraught. I have no particular answers.